zkvm/src/blockchain/state.rs

```diff
@@ -45,19 +45,83 @@ impl BlockchainState {
     fn apply_txlog(&mut self, txlog: &TxLog) -> Result<(), VMError> {
         for entry in txlog.iter() {
             match entry {
-                Entry::Input(contractID) => {
-                    // xxx check contractID is in self.utxos
-                    // xxx remove contractID
+                // Remove input from UTXO set
+                Entry::Input(input) => {
+                    let utxo = input.as_utxo();
+                    if self.utxos.contains(&utxo) {
```

Comment:
Is there a more Rust-idiomatic way to do this whole if/else block? Something like:

`self.utxos.remove_or_die(&utxo)?`

The error returned by `remove_or_die` would have to be converted to a `VMError::InvalidInput`. Is there an idiom for that? If not, maybe this code is good the way it is.

Reply:
Good question! There might very well be such a thing